### PR TITLE
e2e: performance profile: add logs and minor fixes

### PR DESCRIPTION
**Pull request by shajmakh**
Latency testing suite is failing but there is hardly useful info that is logged out to the console.
Add more logs to clarify the steps that are being executed to enable better debugging, along with some minor fixes.

### CODE REVIEW

File: test/e2e/performanceprofile/functests/5_latency_testing/5_latency_testing_suite_test.go [hunk 1]
```diff
@@ -49,6 +49,7 @@ var _ = BeforeSuite(func() {
 	var err error
 	profile, err = profiles.GetByNodeLabels(testutils.NodeSelectorLabels)
 	Expect(err).ToNot(HaveOccurred())
+	testlog.Infof("verify if the isolated cpus value under the performance profile %q is appropriate for this test suite", profile.Name)
```
Comment (fromanirh) [inline test/e2e/performanceprofile/functests/5_latency_testing/5_latency_testing_suite_test.go:52]: this looks like more suitable for a `ginkgo.By`

File: test/e2e/performanceprofile/functests/5_latency_testing/5_latency_testing_suite_test.go [hunk 2]
```diff
@@ -68,7 +69,12 @@ var _ = BeforeSuite(func() {
 			if err != nil {
 				testlog.Error("could not update the profile with the desired CPUs sets")
 			}
+		} else {
+			testlog.Info("current performance profile configuration is valid for the suite, no need to reconfigure it")
 		}
+	} else {
+		testlog.Warning("found 0 nodes with sufficient cpus for the performance profile configuration. Quiting ..")
+		os.Exit(1)
```
Comment (fromanirh) [inline test/e2e/performanceprofile/functests/5_latency_testing/5_latency_testing_suite_test.go:77]: never use `os.Exit` in tests please. better to fail OR to check the prerequisites earlier and act accordingly
Additionally, can we avoid `else`s? :)

File: test/e2e/performanceprofile/functests/utils/profilesupdate/profile_update.go [hunk 3]
```diff
@@ -34,7 +35,7 @@ func UpdateIsolatedReservedCpus(isolatedSet performancev2.CPUSet, reservedSet pe
 
 	err = ApplyProfile(updatedProfile)
 	if err == nil {
-		testlog.Infof("successfully updated performance profile %q with new isolated cpus set: %q and new reserved cpus set: %q", profile.Name, updatedProfile.Spec.CPU.Isolated, updatedProfile.Spec.CPU.Reserved)
+		testlog.Infof("successfully updated performance profile %q with new isolated cpus set: %q and new reserved cpus set: %q", profile.Name, cpuset.MustParse(string(*updatedProfile.Spec.CPU.Isolated)), cpuset.MustParse(string(*updatedProfile.Spec.CPU.Reserved)))
```
Comment (fromanirh) [inline test/e2e/performanceprofile/functests/utils/profilesupdate/profile_update.go:38]: I'd just deference the pointers here. The value added by `MustParse` is not obvious.
```
testlog.Infof("successfully updated performance profile %q with new isolated cpus set: %q and new reserved cpus set: %q", profile.Name, string(*updatedProfile.Spec.CPU.Isolated), string(*updatedProfile.Spec.CPU.Reserved))
```

File: test/e2e/performanceprofile/functests/utils/profilesupdate/profile_update.go [hunk 4]
```diff
@@ -47,10 +47,10 @@ func ApplyProfile(profile *performancev2.PerformanceProfile) error {
 	key, value := components.GetFirstKeyAndValue(mcpLabel)
 	mcpsByLabel, err := mcps.GetByLabel(key, value)
 	if err != nil {
-		return fmt.Errorf("Failed getting MCP by label key %v value %v: %v", key, value, err)
+		return fmt.Errorf("failed getting MCP by label key %v value %v: %v", key, value, err)
```
Comment (fromanirh) [inline test/e2e/performanceprofile/functests/utils/profilesupdate/profile_update.go:50]: TIL, thanks for the reference!

**Review by fromanirh — COMMENTED** [review]
Nice PR! possible improvements inside

File: test/e2e/performanceprofile/functests/5_latency_testing/5_latency_testing_suite_test.go (hunk 2 above)
Comment (shajmakh) [reply test/e2e/performanceprofile/functests/5_latency_testing/5_latency_testing_suite_test.go:77]: thanks for pointing that out, I'll verify and update the code accordingly.
 here the suite did not update the cluster configuration (yet), any reason why not using os.exit in tests?

File: test/e2e/performanceprofile/functests/5_latency_testing/5_latency_testing_suite_test.go [hunk 5]
```diff
@@ -57,18 +59,19 @@ var _ = BeforeSuite(func() {
...
-			testlog.Info("Update the isolated and reserved cpus sets of the profile")
-			err = profilesupdate.UpdateIsolatedReservedCpus(latencyIsolatedSet, latencyReservedSet)
-			if err != nil {
-				testlog.Error("could not update the profile with the desired CPUs sets")
-			}
-		}
+	Expect(len(nodesWithSufficientCpu)).NotTo(Equal(0), "found 0 nodes with sufficient cpus %d for the performance profile configuration. Quiting ..", totalCpus)
+
+	profNeedsUpdate := *initialIsolated != latencyIsolatedSet || *initialReserved != latencyReservedSet
+	if profNeedsUpdate {
```
Comment (Tal-or) [inline test/e2e/performanceprofile/functests/5_latency_testing/5_latency_testing_suite_test.go:69]: Just inline this boolean

File: test/e2e/performanceprofile/functests/5_latency_testing/5_latency_testing_suite_test.go [hunk 6]
```diff
@@ -57,18 +59,19 @@ var _ = BeforeSuite(func() {
...
+		By("Update the isolated and reserved cpus sets of the profile")
+		err = profilesupdate.UpdateIsolatedReservedCpus(latencyIsolatedSet, latencyReservedSet)
+		Expect(err).ToNot(HaveOccurred(), "could not update the profile with the desired CPUs sets")
+	} else {
```
Comment (Tal-or) [inline test/e2e/performanceprofile/functests/5_latency_testing/5_latency_testing_suite_test.go:73]: You can simplify a bit by removing the else and just log that the configuration is valid.
We know if a change is required by the `By` on line 70

File: test/e2e/performanceprofile/functests/5_latency_testing/5_latency_testing_suite_test.go [hunk 7]
```diff
@@ -57,18 +59,19 @@ var _ = BeforeSuite(func() {
...
 	//before applying the changes verify that there are compute nodes with sufficient cpus
-	if len(nodesWithSufficientCpu) != 0 {
-		if *initialIsolated != latencyIsolatedSet || *initialReserved != latencyReservedSet {
-			testlog.Info("Update the isolated and reserved cpus sets of the profile")
-			err = profilesupdate.UpdateIsolatedReservedCpus(latencyIsolatedSet, latencyReservedSet)
-			if err != nil {
-				testlog.Error("could not update the profile with the desired CPUs sets")
-			}
-		}
+	Expect(len(nodesWithSufficientCpu)).NotTo(Equal(0), "found 0 nodes with sufficient cpus %d for the performance profile configuration. Quiting ..", totalCpus)
```
Comment (Tal-or) [inline test/e2e/performanceprofile/functests/5_latency_testing/5_latency_testing_suite_test.go:66]: `Except` is well known for terminating ginkgo test. You don't need to mention that it's quitting

**Comment by Tal-or** [comment]
/lgtm
/approve

Mostly aesthetic changes. looking good

**Review by fromanirh — COMMENTED** [review]
/lgtm
/approve

**Comment by fromanirh** [comment]
/retest
infra issue

**Comment by fromanirh** [comment]
uhhhm the PAO lines is timing out after ~2h, but that's too strict. Let's check where and why this timeout is set

**Pull request merged**